### PR TITLE
Build lit as part of regular Windows build

### PR DESCRIPTION
**Pull request by itzg**
# Fixes

https://jira.rax.io/browse/CMC-2273

# What

Even with the luvi set to 2.9.3-sigar in `package.lua` the built agent exe was producing an error like https://github.com/luvit/lit/issues/252 .

What ended up fixing this (in local build testing at least) was doing two things:
1. Build lit from source, such as what the Linux builder script does here: https://github.com/virgo-agent-toolkit/rackspace-monitoring-agent-buildbot-builder/blob/master/build.sh#L38-L44 
2. Instead of putting the lit-building logic way over in that repo, just incorporate it into this repo's `make.bat` since it was already doing "get lit" stuff

### CODE REVIEW

File: Make.bat [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 IF NOT "x%1" == "x" GOTO :%1
 
 :rackspace-monitoring-agent
 ECHO "Building agent"
-IF NOT EXIST lit.exe CALL Make.bat lit
-if %errorlevel% neq 0 goto error
 IF NOT EXIST luvi-sigar.exe CALL Make.bat luvi-sigar
 if %errorlevel% neq 0 goto error
+IF NOT EXIST lit.exe CALL Make.bat lit
+if %errorlevel% neq 0 goto error
```
Comment (GeorgeJahad) [inline Make.bat]: why did you move these lines

File: Make.bat [hunk 2]
```diff
@@ -23,14 +24,24 @@ GOTO :end
...
-PowerShell -NoProfile -ExecutionPolicy unrestricted -Command "[Net.ServicePointManager]::SecurityProtocol =  'Tls12'; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/luvit/lit/%LIT_VERSION%/get-lit.ps1'))"
+IF NOT EXIST luvi-sigar.exe CALL Make.bat luvi-sigar
+if %errorlevel% neq 0 goto error
+mkdir build
+cd build
+git clone --recursive https://github.com/luvit/lit.git
+if %errorlevel% neq 0 goto error
+cd lit
+git checkout %LIT_VERSION%
+..\..\luvi-sigar.exe . -- make . ..\..\lit.exe ..\..\luvi-sigar.exe
```
Comment (GeorgeJahad) [inline Make.bat]: it looks like this line is invoking luvi-sigar.exe to build luvi-sigar.exe.  is that correct?

File: Make.bat (hunk 1 above)
Comment (itzg) [reply Make.bat]: There's this weird chicken and egg thing apparently with lit and luvi:

https://github.com/luvit/lit#building-from-source

luvi is actually needed to build since you build `lit` by running its from source via luvi and at the same time provide the luvi exe to be bundled into the lit.

Before the logic was using the "get-lit" script to download a pre-built binary.

File: Make.bat (hunk 2 above)
Comment (itzg) [reply Make.bat]: I should stick a comment in there because I was confused too when reading the "build from source" section of lit's README. Apparently the `make` subcommand of `lit` takes three args:
1. the source directory/zip
2. the output exe filename
3. the luvi to bundle into the output exe

Comment (GeorgeJahad) [reply Make.bat]: that helps, thanks!

**Comment by itzg** [comment]
FYI it took in final testing since the version of PowerShell was older than expected, `reg` wasn't usable, but [found LUVI_ARCH is set](https://github.com/racker/buildbot/blob/17ab133130c4a6853fc2a799650059ae9011fd12/master_agentbuild_cm/agent2.py#L82) in buildbot's definition anyway:

https://github.com/virgo-agent-toolkit/rackspace-monitoring-agent/compare/5a86eda66c9006041a61cdcce5dfaba69ecbfae2...3b9b3818091c0486c8e3fe433ac91ce864dc4fe9
